File: src/main/java/com/core/base/corebase/domain/organization/Announcement.java
package com.core.base.corebase.domain.organization;

import com.core.base.corebase.domain.common.BaseDateTimeEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "announcement")
public class Announcement extends BaseDateTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id", nullable = false)
    private Organization organization;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    protected Announcement() {
        // For JPA
    }

    public Announcement(Organization organization, String title, String content) {
        this.organization = organization;
        this.title = title;
        this.content = content;
    }

    public void update(String title, String content) {
        if (title != null) {
            this.title = title;
        }
        if (content != null) {
            this.content = content;
        }
    }
}

File: src/main/kotlin/com/core/base/corebase/domain/organization/Announcement.kt
package com.core.base.corebase.domain.organization

import jakarta.persistence.*
import net.huray.backend.minuting.entity.common.BaseDateTimeEntity

@Entity(name = "announcement")
class Announcement(
    organization: Organization,
    title: String,
    content: String
): BaseDateTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0L

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id", nullable = false)
    var organization: Organization = organization; protected set

    var title = title; protected set
    var content = content; protected set

    fun update(title: String, content: String) {
        this.title = title
        this.content = content
    }
}